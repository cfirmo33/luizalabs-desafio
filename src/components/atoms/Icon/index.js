import React from 'react'
import PropTypes from 'prop-types'

const Icon = ({ icon }) => (
  <i className={`${icon} icon`} />
)

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
}

export default Icon
