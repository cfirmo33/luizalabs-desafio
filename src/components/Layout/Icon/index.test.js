import React from 'react'
import { shallow } from 'enzyme'
import Icon from './index'

describe('Icon', () => {
  it('Deve renderizar corretamente', () => {
    const wrapper = shallow(
      <Icon icon="fas fa-address-book" />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
