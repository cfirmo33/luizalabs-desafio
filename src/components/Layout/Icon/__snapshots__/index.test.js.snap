// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Deve renderizar corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Icon
    icon="fas fa-address-book"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "fas fa-address-book icon",
    },
    "ref": null,
    "rendered": null,
    "type": "i",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "fas fa-address-book icon",
      },
      "ref": null,
      "rendered": null,
      "type": "i",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
